Source/WebCore/Modules/cookie-store/CookieStore.cpp

```diff
@@ -38,18 +45,45 @@ namespace WebCore {
 
 WTF_MAKE_ISO_ALLOCATED_IMPL(CookieStore);
 
-Ref<CookieStore> CookieStore::create()
+Ref<CookieStore> CookieStore::create(ScriptExecutionContext& context)
 {
-    return adoptRef(*new CookieStore());
+    auto cookieStore = adoptRef(*new CookieStore(context));
+    cookieStore->suspendIfNeeded();
+    return cookieStore;
 }
 
-CookieStore::CookieStore() = default;
+CookieStore::CookieStore(ScriptExecutionContext& context)
+    : ActiveDOMObject(&context)
+{
+}
 
 CookieStore::~CookieStore() = default;
 
-void CookieStore::get(const String&, Ref<DeferredPromise>&& promise)
+void CookieStore::get(const String& name, Ref<DeferredPromise>&& promise)
 {
-    promise->reject(NotSupportedError);
+    auto* context = scriptExecutionContext();
+    auto* origin = context->securityOrigin();
+    if (origin->isOpaque()) {
+        promise->reject(SecurityError);
+        return;
+    }
+
+    auto& url = context->url();
+    auto& document = *downcast<Document>(context);
+    auto& cookieJar = document.page()->cookieJar();
+    auto cookieStoreGetoptions = CookieStoreGetOptions { name, { } };
+    auto completionHandler = [promise = WTFMove(promise)] (Vector<Cookie>&& cookies) {
+        if (cookies.isEmpty()) {
+            promise->resolveWithJSValue(JSC::jsNull());
```

Comment:
I tried other ways of writing this line, but none of them seemed to work, so I kept this.